protocol/x/clob/keeper/liquidations.go

```diff
@@ -138,8 +138,13 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 	for i := 0; i < int(k.Flags.MaxDeleveragingAttemptsPerBlock) && i < len(unfilledLiquidations); i++ {
 		liquidationOrder := unfilledLiquidations[i]
 
-		subaccountId := liquidationOrder.GetSubaccountId()
 		perpetualId := liquidationOrder.MustGetLiquidatedPerpetualId()
+		subaccountId := liquidationOrder.GetSubaccountId()
+
+		subaccount := k.subaccountsKeeper.GetSubaccount(ctx, subaccountId)
+		if _, exists := subaccount.GetPerpetualPositionForId(perpetualId); !exists {
+			continue
+		}
```

Comment:
Should we change [this line](https://github.com/dydxprotocol/v4-chain/blob/77573aaceca5bf83d9372b42b64ccf8f11e24dbf/protocol/x/clob/keeper/deleveraging.go#L56-L57) to a panic as part of this change (since now the position should always be open)?

Reply:
Alternatively, wdyt of passing in `position` to `MaybeDeleverageSubaccount`? Up to you.